plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -338,6 +344,7 @@ public class IcebergMetadata
     private final Map<IcebergTableHandle, TableStatistics> tableStatisticsCache = new ConcurrentHashMap<>();
 
     private Transaction transaction;
+    private Optional<Map<SchemaTableName, Long>> fromSnapshotsForRefresh = Optional.empty();
```

Comment:
This can be simplified to just be `Optional<Long>` since we support single source table only.
We could maybe have separate `boolean refreshInProgress`, but IMO it's not needed